lib/pbench/server/api/resources/__init__.py

```diff
@@ -155,7 +155,7 @@ def __str__(self) -> str:
 class SchemaError(APIAbort):
     """Generic base class for errors in processing a JSON schema."""
 
-    def __init__(self, http_status: int = HTTPStatus.BAD_REQUEST):
+    def __init__(self, http_status: int = HTTPStatus.BAD_REQUEST, **_kwargs):
```

Comment:
What's the point of adding an unused `kwargs`??

Reply:
Ugh; this is ugly. OK, so `ConversionError` is designed with `**kwargs` to allow callers to pass an `http_status` *through* the `ConversionError` constructor to `SchemaError` ... but if anyone used other `kwargs` it wouldn't work, and the explicit declaration here is misleading because `SchemaError` really doesn't support kwargs. (Pass-through or otherwise.)

I'm curious where you're seeing a lint warning, because I'm not. 😦

Reply:
FYI, the `http_status` is only specified through `ConversionError` in `convert_username` and `test_exceptions.py` ... I don't see any lint annotations. I'm curious exactly what you're seeing, and this workaround bugs me because it shouldn't be necessary.

Reply:
Looking at the `main` branch, I'm getting `Unexpected argument` on the `super().__init__()` invocation in `DatasetConversionError.__init__()`, but, interestingly, I _don't_ get the complaint for the one in `ConversionError.__init__()`.  🤷

Reply:
Sure, I guess, because `DatasetConversionError` passes `http_status` explicitly and doesn't actually use `kwargs`. (In fact, it can't, aside from repeating the `http_status`, because it's not good for anything else.)

I think your linter is broken. But if you really want to quiet its ranting, a better "ugly workaround" would be to add an explicit `http_status: Optional[int] = None` to `ConversionError`, pass it on, and drop the `kwargs` entirely from both `ConversionError` and `DatasetConversionError` (which, so far as I can see, is never used anyway... maybe it was at one time).

But again, I don't think there's anything illegal or even unreasonable here, and I think this is just an argument against your linter. 😆

Reply:
I can't really argue _against_ my linter -- it is more or less built-in to my IDE; and arguing _with_ a linter is, as we know, largely pointless.  🙃 

At this point, I'm not inclined to disturb the `SchemaError` (and children) code any further.  Given that the Production deployment seems happy, shall I go ahead and merge this?

Reply:
I really dislike adding `kwargs` here, because it makes no sense and is potentially misleading. But, whatever: it'll never be perfect.